### PR TITLE
migrate adapter to asto-0.15 version

### PR DESCRIPTION
**Pull request by brastak**
For #8

### CODE REVIEW

File: src/main/java/com/artipie/goproxy/Goproxy.java [hunk 1]
```diff
@@ -164,54 +167,55 @@ private Completable actualUpdate(final String repo,
...
-                        new ListOf<>(
-                            new String(Files.readAllBytes(list)).split("\n")
-                        ),
-                        new ListOf<>(String.format("v%s", version))
-                    )
+    private static Single<Content> updateFileList(final String version, final Content content) {
+        return Flowable.fromPublisher(content)
+            .collectInto(
+                ByteBuffer.allocate(0),
+                (left, right) -> ByteBuffer.allocate(left.remaining() + right.remaining())
```
Comment (Vatavuk) [inline src/main/java/com/artipie/goproxy/Goproxy.java]: @brastak extract this to a private method to make it more obvious (declarative) what are we doing here.

File: src/main/java/com/artipie/goproxy/Goproxy.java [hunk 2]
```diff
@@ -164,54 +167,55 @@ private Completable actualUpdate(final String repo,
...
+                    .put(left).put(right)
+                    .flip()
+            ).map(
+                buf -> new Remaining(buf).bytes()
+            ).map(
+                buf -> new org.cactoos.text.Joined(
+                "\n",
+                new Joined<String>(
```
Comment (Vatavuk) [inline src/main/java/com/artipie/goproxy/Goproxy.java]: @brastak extract this to a private method to make it more obvious (declarative) what are we doing here.

**Comment by 0crat** [comment]
Job [#13](https://github.com/artipie/go-adapter/issues/13) is already in scope

**Comment by Vatavuk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/go-adapter/issues/13#issuecomment-599737967))

@Vatavuk This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#13](https://github.com/artipie/go-adapter/issues/13) is [in scope](http://www.zerocracy.com/policy.html#14) for 5hr
  * The role is `REV`
  * The job is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk) for [5hr](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @vatavuk[/z](https://www.0crat.com/u/vatavuk) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @brastak[/z](https://www.0crat.com/u/brastak): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fgo-adapter%2313%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
This pull request [#13](https://github.com/artipie/go-adapter/issues/13) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/7a2aadbc-0411-4b32-9161-6788b56492d9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/goproxy/Goproxy.java (hunk 2 above)
Comment (brastak) [reply src/main/java/com/artipie/goproxy/Goproxy.java]: @Vatavuk please, review update.

**Comment by brastak** [comment]
@Vatavuk please, review update

**Comment by 0crat** [comment]
Job was finished in 22 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Pull request merged**